src/Formik.tsx

```diff
@@ -460,7 +474,7 @@ export function useFormik<Values extends FormikValues = FormikValues>({
     (values: Values) => {
       dispatch({ type: 'SET_VALUES', payload: values });
       return validateOnChange
-        ? validateFormWithLowPriority(state.values)
+        ? debouncedValidateFormWithLowPriority(state.values)
```

Comment:
This is really an obvious change, dispatch will never update `state` object synchronously.
```suggestion
        ? debouncedValidateFormWithLowPriority(values)
```